src/main/java/com/jprobstats/dataset/impl/CompositeList.java

```diff
@@ -0,0 +1,44 @@
+package com.jprobstats.dataset.impl;
+
+import com.jprobstats.dataset.JProbStatsList;
+import com.jprobstats.dataset.ListFactory;
+/**
+ * Combining two JProbStatsLists
+ *
+ * @param <T>
+ */
+public class CompositeList<T> extends JProbStatsList<T> {
+	public final JProbStatsList<T> front;
+	public final JProbStatsList<T> back;
+	private final int size;
+
+	private CompositeList(JProbStatsList<T> a, JProbStatsList<T> b) {
+		front = a;
+		back = b;
+		size = a.size() + b.size();
+	}
+	// TODO when both list size exceeds tuple size we have to increase destination array.
+	public static <T> JProbStatsList<T> concat(JProbStatsList<T> a, JProbStatsList<T> b) {
```

Comment:
If my dataset is null list then size will be 0.it wont throw null ponter exception.